src/libraries/System.Text.RegularExpressions/gen/RegexAnalyzer.cs

```diff
@@ -0,0 +1,416 @@
+        private const string TimeSpanTypeName = "System.TimeSpan";
+        private const string TimeoutTypeName = "System.Threading.Timeout";
+
+        // internal members
+        internal const string PatternIndexName = "PatternIndex";
+        internal const string RegexOptionsIndexName = "RegexOptionsIndex";
+        internal const string RegexTimeoutIndexName = "RegexTimeoutIndex";
+        internal const string RegexTimeoutName = "RegexTimeout";
+        internal const string DiagnosticId = "SYSLIB1046";
+
+        /// <inheritdoc />
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(DiagnosticDescriptors.UseRegexSourceGeneration);
+
+        /// <inheritdoc />
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+
+            // Register analysis of calls to the Regex constructors
+            context.RegisterOperationAction(AnalyzeObjectCreation, OperationKind.ObjectCreation);
+
+            // Register analysis of calls to Regex static methods
+            context.RegisterOperationAction(AnalyzeInvocation, OperationKind.Invocation);
+        }
+
+        /// <summary>
+        /// Analyzes an invocation expression to see if the invocation is a call to one of the Regex static methods,
+        /// and checks if they could be using the source generator instead.
+        /// </summary>
+        /// <param name="context">The operation context representing the invocation.</param>
+        private void AnalyzeInvocation(OperationAnalysisContext context)
+        {
+            // Ensure source generator is supported.
+            if (!ProjectSupportsRegexSourceGenerator(context.Operation))
+            {
+                return;
+            }
+
+            // Ensure the invocation is a Regex static method.
+            IInvocationOperation invocationOperation = (IInvocationOperation)context.Operation;
+            IMethodSymbol method = invocationOperation.TargetMethod;
+            if (!method.IsStatic || !SymbolEqualityComparer.Default.Equals(method.ContainingType, context.Compilation.GetTypeByMetadataName(RegexTypeName)))
+            {
+                return;
+            }
+
+            // Depending on the static method being called, we need to save the parameters as properties so that we can save them onto the diagnostic so that the
+            // code fixer can later use that property bag to generate the code fix and emit the RegexGenerator attribute.
+            // Most static methods have the same parameter overloads which are covered by the first if block. Replace static method takes extra parameters so that one
+            // is treated specially.
+            if (method.Name is "IsMatch" or "Match" or "Matches" or "Split" or "Count" or "EnumerateMatches")
```

Comment:
We don't recommend comparing by symbol names. Instead it is recommended that in the prior check you fetch all the relevant method symbols on the Regex type, add them to a hashset and then directly compare if the target method is part of that set. The code to fetch all the well-known types and methods should be moved to CompilationAction instead of doing it in OperationAction.

Reply:
I see, I will fix this.